scenariolib/event_searchandclick.go

```diff
@@ -31,8 +33,30 @@ func newSearchAndClickEvent(e *JSONEvent) (*SearchAndClickEvent, error) {
 		return nil, errors.New("Parameter queryText must be of type string in SearchAndClickEvent")
 	}
 
-	if docClickTitle, validCast = e.Arguments["docClickTitle"].(string); !validCast {
-		return nil, errors.New("Parameter docClickTitle must be of type string in SearchAndClickEvent")
+	if e.Arguments["docClickTitle"] != nil {
+		if docClickTitle, validCast = e.Arguments["docClickTitle"].(string); !validCast {
+			docClickTitle = ""
+		}
+	}
+	if e.Arguments["matchField"] != nil {
+		if matchField, validCast = e.Arguments["matchField"].(string); !validCast {
+			matchField = ""
+		}
+	}
+	if e.Arguments["matchValue"] != nil {
+		if matchValue, validCast = e.Arguments["matchValue"].(string); !validCast {
+			matchValue = ""
+		}
+	}
+
+	if matchField != "" && matchValue == "" {
```

Comment:
Hmmm the field to match could be an empty string no?
@author == "" 
would be a valid query?